src/Lumper.UI/Updater/Updater.cs

```diff
@@ -0,0 +1,218 @@
+            ProcessStartInfo processInfo = new ProcessStartInfo();
+            Process process;
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                processInfo = new ProcessStartInfo("cmd.exe", "/c " + command);
+                processInfo.CreateNoWindow = true;
+                processInfo.UseShellExecute = false;
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                processInfo = new ProcessStartInfo(command);
+                processInfo.CreateNoWindow = true;
+                processInfo.UseShellExecute = true;
+            }
+            processInfo.RedirectStandardError = true;
+            processInfo.RedirectStandardOutput = true;
+            processInfo.WorkingDirectory = System.IO.Directory.GetCurrentDirectory();
+            process = Process.Start(processInfo);
+        }
+
+        //Grab the JSON from the github API and deserialize
+        private static async Task<GHUpdate> GetGithubUpdates()
+        {
+            HttpClient client = new HttpClient();
+
+            var request = new HttpRequestMessage()
+            {
+                RequestUri = new Uri("https://api.github.com/repos/momentum-mod/lumper/releases"),
+                Method = HttpMethod.Get
+            };
+            client.DefaultRequestHeaders.Add("User-Agent", "Other");
+            HttpResponseMessage m = client.Send(request);
+
+            if (m.IsSuccessStatusCode)
+            {
+                string response = await m.Content.ReadAsStringAsync();
+
+                var assets = JsonConvert.DeserializeObject<GHUpdate>(JsonObject.Parse(response)[0].ToString());
+                return assets;
+            }
+            else
+                throw new Exception("Could not connect - error " + m.StatusCode);
+        }
+        
+        public static async Task<bool> CheckForUpdate()
+        {
+            GHUpdate assets = await GetGithubUpdates();
+
+            //parse tag name to find the current and latest version
+            //finding the format of xx.yy.zz
+            string newestVersionSplit = Regex.Match(assets.tag_name, "[0-9]+\\.[0-9]+\\.[0-9]+").ToString();
+            MMP current = GetMMPVersion(Assembly.GetExecutingAssembly().GetName().Version.ToString());
+            MMP latest = GetMMPVersion(newestVersionSplit);
+
+            return (current.major != latest.major || current.minor != latest.minor || current.patch != latest.patch);
+        }
+        //returns the URL to the download link for the OS-specific version
+        private static string GetPath(GHUpdate assets, string OS)
+        {
+            for (int i = 0; i < assets.assets.Length; i++)
+            {
+                if (assets.assets[i].name.ToLower().Contains(OS.ToLower()))
+                {
+                    return assets.assets[i].browser_download_url;
+                }
+
+            }
+            return null;
+        }

```

Comment:
Use a `foreach` loop, also can use a StringComparison overload:
```cs
foreach (Asset asset in githubUpdate.assets)
    if (asset.Name.Contains(OS, StringComparison.OrdinalgnoreCase)
        return asset.BrowserDownloadUrl;

return null;
```

Note that method method can return `null`, so needs to be annotated that what, then you should check for that in below code.